src/client.d.ts

```diff
@@ -0,0 +1,294 @@
+                provide(name: string, callback: (response: RPCResponse) => void): void;
+                /**Removes the client as a provider previously registered using provide(). */
+                unprovide(name: string): void;
+                /**
+                 * Executes a remote procedure call. callback will be invoked with the returned result or with an error if the RPC failed.
+                 */
+                make(name: string, data: any, callback: Function): void;
+        }
+
+        interface EventEmitter {
+                /**Subscribe to an event. */
+                on(type: string, callback: Function): void;
+                /**Unsubscribes from an event by:
+        - removing a specific callback when called with both arguments.
+        deepstream.off( 'error', errorCallback )
+        - removing all listeners for an event when only called with an event.
+        deepstream.off( 'error' )
+        - removing all listeners for all events if called without arguments.
+        deepstream.off() */
+                off(type?: string, callback?: Function): void;
+                /** Register a one-time listener for an event. The listener will be removed immediately after its first execution. */
+                once(type: string, callback: Function): void;
+                /**Emits an event. */
+                emit(event: string, arguments?: any): void;
+                /**Returns an array of listeners that are registered for the event. */
+                listeners(event: string): any[];
+                /**Returns true if there are listeners registered for that event. */
+                hasListeners(event: string): boolean;
+                /**Authenticates the client against the server. To learn more about how authentication works, please have a look at the Security Overview.
+                 *Callback will be called with: success (Boolean), data (Object).
+                 */
+        }
+
+        interface Client extends EventEmitter {
+
+                login(authParams?: {}, callback?: Function): Client;
+                /**Closes the connection to the server. */
+                close(): void;
+                /**Returns the current connectionState. Please find a list of available connectionStates here. */
+                getConnectionState(): ConnectionState;
+                /**Returnes a unique id. The uid starts with a Base64 encoded timestamp to allow for semi-sequentual ordering and ends with a random string. */
+                getUid(): string;
+
+                record: RecordStatic;
+
+                event: EventStatic;
+
+                rpc: RPCStatic;
+        }
+
+        interface deepstreamQuarantine extends Client {
+
+        }
+        interface deepstreamStatic {
+                CONSTANTS: {
+                        [key: string]: string
```

Comment:
That works. We'll also want to update the enums to use strings. Typescript enums compile to integers.